### PR TITLE
Add product section

### PR DESCRIPTION
**Pull request by dhmacs**
This PR adds the first section of the product page with the following features:

- Product gallery (that is in sync with the product options selection)
- Title and pricing
- Product options
- Add to cart (the button doesn't automatically open the cart drawer, this will be done in a separate PR)
- Inventory details
- Product description 
- Kit content
- Product spec
- Warranty
- Prop 65 warnings

The section is not responsive yet, and it's not necessarily super polished. The aim of this PR is to merge the bulk of it, and gradually refine it in follow up PR.

## QA

1. Visit Vercel Preview
2. Go to [/Product/iphone-11-pro-screen](https://react-commerce-git-add-product-section-ifixit.vercel.app/Products/iphone-11-pro-screen)
3. Verify that the features described above are present

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

All of the features listed above are present 👍 

### Testing Procedure:
1. Navigate to [/iphone-11-pro-screen](https://react-commerce-git-add-product-section-ifixit.vercel.app/Products/iphone-11-pro-screen)
2. Ensure that each feature shown above is present
3. Note any bugs present

---

### Demo:

https://user-images.githubusercontent.com/95656772/186731471-1f516e03-f090-46de-87d9-c85524dcbf93.mp4

---

### Additional Notes:

There are some things that I noticed that I wanted to take note of for possible refinements:

>1. The **Description** and **Specifications** drawers are open initially (0:47 in demo)

>2. The yellow info box can be formatted better, maybe a bullet list or something to make it more readable (0:52 in demo)

>3. Extra horizontal divider inside **Specifications** drawer (1:50 in demo)

>4. The **Display Panel** description changes when the LCD variant is selected between _Fix Kit_ and _Part Only_ (1:54 in demo)

>5. No tip on the information button ℹ️ for Shipping Restrictions (2:06 in demo)

>6. Prop 65 warning link isn't direct, the placement for this warning could also be placed more away from the footer (2:15 in demo)

>7. Changing the condition when Part Only is selected will remove the Fix Kit image and also make it unclickable, this also removes the image preview container (2:30 in demo)

**Comment by jordycosta** [comment]
QA 🟢 Carry-over

https://github.com/iFixit/react-commerce/pull/650#issuecomment-1227581087

File: frontend/templates/product/sections/ProductSection/ProductOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      </Box>
+   );
+}
+
+type ProductOptionImageProps = {
+   image: Image;
+};
+
+function ProductOptionImage({ image }: ProductOptionImageProps) {
+   const ratio =
```
Comment (masonmcelvain) [inline frontend/templates/product/sections/ProductSection/ProductOptions.tsx:196]: Is it intentional that we're not using the `ratio` here, besides the null check?

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Looks great! deploy_block ✌🏻 on a question that can be addressed in a followup pull if it requires a code change.

File: frontend/templates/product/sections/ProductSection/ProductOptions.tsx (hunk 1 above)
Comment (dhmacs) [reply frontend/templates/product/sections/ProductSection/ProductOptions.tsx:196]: Oh good question. It's probably a bad refactor 🤔 Anyway next week @federicobadini will run through these to make them responsive, so I'll let him address this as he sees fit.

un_deploy_block 🏎️

**Pull request merged**